alf/nest/utils.py

```diff
@@ -207,10 +221,14 @@ def _convert_cpu(tensor):
         else:
             return tensor
 
-    d = alf.get_default_device()
+    if device is None:
+        d = alf.get_default_device()
+    else:
+        d = device
+
     if d == 'cpu':
         return nest.map_structure(_convert_cpu, nests)
-    elif d == 'cuda':
+    elif d == 'cuda' and torch.cuda.is_available():
```

Comment:
cuda.is_available() should be inside this branch and raise error otherwise

Reply:
Done